src/Tree/Entity/Repository/NestedTreeRepository.php

```diff
@@ -172,24 +187,69 @@ public function getPathQueryBuilder($node)
      * Get the Tree path query by given $node
      *
      * @param object $node
+     * @phpstan-param array{includeNode?: bool} $options
+     *
+     * $options['includeNode'] = (bool) Whether to include the node itself (default: true)
      *
-     * @return \Doctrine\ORM\Query
+     * @return Query
      */
-    public function getPathQuery($node)
+    public function getPathQuery($node, array $options = [])
```

Comment:
This is a BC break.

Reply:
I'm sorry, how is this a BC break?

Reply:
This is adding a new parameter to a method that (currently) can be extended by users.
See:
* https://3v4l.org/ldmIb;
* https://en.wikipedia.org/wiki/Liskov_substitution_principle.

Reply:
Well yes, you have a point, but if we're going to consider this a breaking change, I have also added new methods, such as `getPathAsString()` and `recoverFast()`, so they are then breaking changes as well. What do you suggest to do about that?

Reply:
AFAIK, adding new methods in a class that is not part of the current API is not a BC break.

Reply:
you can do something similar to: 

```php
$options = func_get_args()[1] ?? [];

if (!\is_array($options)) {
    throw new TypeError('Argument 2 MUST be an array');
}
```
And we'll add the parameter in the next major version.

Reply:
Are you suggesting that I create a separate PR where the only changes are adding the above snippet in the following methods?
```
    public function getPathQueryBuilder($node)
    public function getPathQuery($node)
    public function getPath($node)
    public function verify()
    public function recover()
```
...and then this PR will just wait for a next major version?

TBH, I'm not sure what help will this snippet do. Someone might have extended the method and added a second array argument, in which case they won't get a TypeError and in the next major version, their code will still be broken.

Reply:
Apologies, I've been away for a while. I've made the requested changes now.